core/GlobalState.cc

```diff
@@ -1327,6 +1327,11 @@ MethodRef GlobalState::enterMethodSymbol(Loc loc, ClassOrModuleRef owner, NameRe
     if (store.exists()) {
         ENFORCE(store.isMethod(), "existing symbol is not a method");
         counterInc("symbols.hit");
+        if (!store.locs(*this).empty() && absl::c_count(store.locs(*this), loc) == 0) {
```

Comment:
Can you explain what your goal with this condition is?

My guess would have been that we would not have needed any condition here.